src/main/java/io/stargate/sgv2/jsonapi/service/operation/filters/table/codecs/VectorCodecs.java

```diff
@@ -0,0 +1,82 @@
+package io.stargate.sgv2.jsonapi.service.operation.filters.table.codecs;
+
+import com.datastax.oss.driver.api.core.data.CqlVector;
+import com.datastax.oss.driver.api.core.type.VectorType;
+import com.datastax.oss.driver.api.core.type.reflect.GenericType;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import io.stargate.sgv2.jsonapi.api.model.command.clause.filter.JsonLiteral;
+import io.stargate.sgv2.jsonapi.exception.catchable.ToCQLCodecException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+/**
+ * Container for factories of codecs that handle CQL Vector type. Separated from main {@link
+ * JSONCodecs} to keep the code somewhat modular.
+ */
+public abstract class VectorCodecs {
+  private static final GenericType<List<Float>> FLOAT_LIST = GenericType.listOf(Float.class);
+
+  public static <JavaT, CqlT> JSONCodec<JavaT, CqlT> arrayToCQLFloatVectorCodec(
+      VectorType vectorType) {
+    // Unfortunately we cannot simply construct and return a single Codec instance here
+    // because VectorType's dimensions vary, and we need to know the expected dimensions
+    // (unless we want to rely on DB validating dimension as part of write and catch failure)
+    return (JSONCodec<JavaT, CqlT>)
+        new JSONCodec<>(
+            FLOAT_LIST,
+            vectorType,
+            (cqlType, value) -> toCQLFloatVector(vectorType, value),
+            // This codec only for to-cql case, not to-json, so we don't need this
+            null);
+  }
+
+  public static <JavaT, CqlT> JSONCodec<JavaT, CqlT> toJSONFloatVectorCodec(VectorType vectorType) {
+    return (JSONCodec<JavaT, CqlT>)
+        new JSONCodec<>(
+            FLOAT_LIST,
+            vectorType,
+            // This codec only for to-json case, not to-cql, so we don't need this
+            null,
+            (objectMapper, cqlType, value) -> toJsonNode(objectMapper, (CqlVector<Number>) value));
+  }
+
+  static CqlVector<Float> toCQLFloatVector(VectorType vectorType, Collection<?> listValue)
+      throws ToCQLCodecException {
+    Collection<JsonLiteral<?>> vectorIn = (Collection<JsonLiteral<?>>) listValue;
+    final int expLen = vectorType.getDimensions();
+    if (expLen != vectorIn.size()) {
+      throw new ToCQLCodecException(
+          vectorIn,
```

Comment:
Just wild thinking, does this mean, the error msg has a long user-input vector in the response?
Do we want to just let user know there is a dimension mismatch.

Reply:
I think this needs to be handled generically, to limit output length for all kinds of input values: I think `ToCQLCodecException` expects input value to be provided.
I'll see if `ToCQLCodecException` limits length yet.

Reply:
Ok: value was NOT yet truncated -- added code to truncate to 1000 chars if longer.